interp/builtin.go

```diff
@@ -911,6 +911,14 @@ func (p *flagParser) more() bool {
 		p.remaining = p.remaining[1:]
 		return false
 	}
+	if arg == "-" {
+		if len(p.remaining) > 1 {
+			p.remaining = p.remaining[1:]
+			return false
+		}
+		p.remaining = nil
+		return false
+	}
```

Comment:
The `-` argument is only special in `set`, as far as I can tell, as it's described in `help set` but not the more generic `man getopts` which is the style that most builtins follow.

So I'd say the special case of `-` needs to be handled in the `set` builtin, not here.